src/Handler/CurlFactory.php

```diff
@@ -142,12 +145,19 @@ private static function invokeStats(EasyHandle $easy): void
     private static function finishError(callable $handler, EasyHandle $easy, CurlFactoryInterface $factory): PromiseInterface
     {
         // Get error information and release the handle to the factory.
+        /** @var array<array-key, mixed> $curlStats */
+        $curlStats = \curl_getinfo($easy->handle);
+
         $ctx = [
             'errno' => $easy->errno,
             'error' => \curl_error($easy->handle),
             'appconnect_time' => \curl_getinfo($easy->handle, \CURLINFO_APPCONNECT_TIME),
-        ] + \curl_getinfo($easy->handle);
-        $ctx[self::CURL_VERSION_STR] = \curl_version()['version'];
+        ] + $curlStats;
+
+        /** @var array<array-key, mixed> $curlVersion */
+        $curlVersion = \curl_version();
```

Comment:
PHPStan and PSalm seems to disagree here, where PHPStan will correctly assume that both `\curl_getinfo()` and `\curl_getversion()` can return `false`.

`\curl_getinfo()`:
This seems to be an issue with the PHPStan stubs where `false` is only plausible if both arguments is passed to `\curl_getinfo()` (which we don't here): https://github.com/php/php-src/blob/master/ext/curl/interface.c#L3360

`\curl_version()`:
PHP has a check to see if cURL returns a null pointer, however the structure returned by atleast more modern versions of cURL:
 - `php-src` commit which introduced this check: https://github.com/php/php-src/commit/537284418cde811f6541708a928430c7967d4a27#diff-29336a112e08c5b5caa7950afb6601010867f40a4643a33263646504277f7b33R612
 - `php-src`: https://github.com/php/php-src/blob/master/ext/curl/interface.c#L1691
 - `curl`: https://github.com/curl/curl/blob/63c76681827b5ae9017f6c981003cd75e5f127de/lib/version.c#L503